packages/shared/src/components/buttons/ToggleClickbaitShield.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { type ReactElement } from 'react';
+import { Button, ButtonSize, ButtonVariant } from './Button';
+import { ShieldCheckIcon, ShieldIcon, ShieldPlusIcon } from '../icons';
+import { useSettingsContext } from '../../contexts/SettingsContext';
+import { usePlusSubscription } from '../../hooks';
+import { SidebarSettingsFlags } from '../../graphql/settings';
+import { useLogContext } from '../../contexts/LogContext';
+import { LogEvent, TargetId } from '../../lib/log';
+import { useLazyModal } from '../../hooks/useLazyModal';
+import { SimpleTooltip } from '../tooltips';
+import { LazyModal } from '../modals/common/types';
+import { FilterMenuTitle } from '../filters/helpers';
+
+export const ToggleClickbaitShield = (): ReactElement => {
+  const { openModal } = useLazyModal();
+  const { isPlus } = usePlusSubscription();
+  const { logEvent } = useLogContext();
+  const { flags, updateFlag } = useSettingsContext();
+
+  if (!isPlus) {
+    return (
+      <SimpleTooltip
+        placement="bottom"
+        content="Enable Clickbait Shield to get clearer more informative titles"
+        container={{
+          className: 'max-w-64 text-center',
+        }}
+      >
+        <Button
+          size={ButtonSize.Medium}
+          variant={ButtonVariant.Float}
+          icon={<ShieldPlusIcon />}
```

Comment:
Design states hover should be secondary icon :jenkins-triggered:
I think I did it somewhere else, can't fully recall how I fixed it there though.
Maybe it can be some addition to button?

![Screenshot 2024-12-04 at 14 20 38](https://github.com/user-attachments/assets/6c0d809b-f90d-4eaa-8233-4c944d9ae39e)

Reply:
:triggered: yeah.. forgo that

Reply:
Added possibility to use secondary icon when hovering buttons :)
Also when using secondary version of icon, it uses the primary version on hover instead.